quint/src/importResolver.ts

```diff
@@ -170,4 +172,89 @@ class ImportResolverVisitor implements IRVisitor {
       typeDefs.forEach(def => addTypeToTable(def, this.table))
     }
   }
+
+
+  // Imported names are copied with a scope since imports are not transitive by
+  // default. Exporting needs to turn those names into unscoped ones so, when
+  // the current module is imported, the names are accessible. Note that it is
+  // also possible to export names that were not imported.
+  enterExport(def: QuintExport) {
+    if (def.protoName === this.currentModule?.name) {
+      // Exporting current module
+      this.errors.set(def.id, {
+        code: 'QNT407',
+        message: `Cannot export ${def.protoName} inside ${def.protoName}`,
+        data: {},
+      })
+      return
+    }
+
+    const moduleTable = this.tables.get(def.protoName)
+    if (!moduleTable) {
+      // Exporting unexisting module
+      this.errors.set(def.id, {
+        code: 'QNT404',
```

Comment:
Nice error code :)

Reply:
It's from your initial ADR, so kudos to you :smile: